### PR TITLE
Add memory types

### LINKED ISSUE #20: Implement Memory Types
https://webassembly.github.io/spec/core/text/types.html#memory-types
https://webassembly.github.io/spec/core/syntax/types.html#memory-types

### PR DESCRIPTION
**Pull request by welnanick**
Resolves #20

### CODE REVIEW

File: library/src/main/java/kwasm/format/text/Type.kt [hunk 1]
```diff
@@ -135,13 +138,17 @@ sealed class Type<T>(
         }
     }
 
+    /**
+     * From https://webassembly.github.io/spec/core/text/types.html#memory-types:
```
Comment (jasonwyatt) [inline library/src/main/java/kwasm/format/text/Type.kt]: Can you make this a link, like in the other KDocs?

File: library/src/main/java/kwasm/format/text/Type.kt [hunk 2]
```diff
@@ -135,13 +138,17 @@ sealed class Type<T>(
...
+     * ```
+     */
     class MemoryType(
         sequence: CharSequence,
         context: ParseContext? = null
-    ) : Type<Unit>(sequence, context) {
-        override fun parseValue() {
-            TODO("not implemented")
-        }
+    ) : Type<Limits>(sequence, context) {
```
Comment (jasonwyatt) [inline library/src/main/java/kwasm/format/text/Type.kt]: Should `Memory` be its own `ast` class?

```kotlin
package kwasm.ast

data class Memory(val limits: Limits)
```

Comment (welnaseth) [reply library/src/main/java/kwasm/format/text/Type.kt]: +1

File: library/src/main/java/kwasm/format/text/Type.kt (hunk 1 above)
Comment (welnanick) [reply library/src/main/java/kwasm/format/text/Type.kt]: Done

File: library/src/main/java/kwasm/format/text/Type.kt (hunk 2 above)
Comment (welnanick) [reply library/src/main/java/kwasm/format/text/Type.kt]: Done

File: library/src/main/java/kwasm/ast/Memory.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ * either express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kwasm.ast
+
+import kwasm.format.text.Type
+
+/** This data class is used to represent the structure of a [Type.MemoryType] */
+data class Memory(val limits: Type.Limits)
```
Comment (welnanick) [inline library/src/main/java/kwasm/ast/Memory.kt]: Actually wait a minute. I shouldn't do this. The ast should be independent of the format of WASM we're reading in. I gotta think how I want to reconfigure this.

Comment (jasonwyatt) [reply library/src/main/java/kwasm/ast/Memory.kt]: There should be a Limits ast value too.

Comment (welnanick) [reply library/src/main/java/kwasm/ast/Memory.kt]: Done

**Pull request merged**